Assingmenty/DiceRollGame/GameManager.cs

```diff
@@ -0,0 +1,34 @@
+﻿class GameManager
+{
+    private readonly int _randomNumber = RandomNumberProvider.Generate();
```

Comment:
Klasy stateless są zawsze lepsze niż stateful (łatwiejsze rozumienie i testowanie kodu). Tutaj nie ma potrzeby żeby te zmienne były prywatne w klasie. Powinny być lokalne dla metody run. Teraz np gdybyś odpaliła grę kilka razy, to za każdym razem będziesz mieć ten sam _randomNumber.

Druga uwaga to że ze statycznym RandomNumberProviderem testowanie jest niemożliwe. Nie da się zrobić dependency injection statycznej klasy, więc jej nie zamockujemy. teraz to może jeszcze nie takie ważne, bo jeszcze nie uczyłaś się testowania.

Reply:
No dobra, ale ze zmiennej _numberOfTries korzystają obie metody w Game. Czy w takim razie lepiej tą zmienną zrobić lokalną dla Run() i przekazać ją do PlayGame(), czy jednak zrobić z niej pole klasy?
![image](https://user-images.githubusercontent.com/15715692/204274387-8b75e210-75fe-4c82-a3ef-0773c0b76532.png)

Reply:
Dołożyłam jeszcze _diceSides po to, żeby funkcja generujące wynik rzutu kostką była bardziej elastyczna i tym bardziej wydaje mi się, że _numberOfTries i _diceSides to cechy Gry i dlatego powinny być polami klasy. Ale jestem ciekawa twojego zdania.
![image](https://user-images.githubusercontent.com/15715692/204279029-b0c3df2b-a6d8-48fc-9f60-37c93b1d1480.png)

Reply:
W sumie masz rację, number of tries jest cechą gry i może należeć do tej klasy i być jej polem. Aczkolwiek. ja bym zostawiła jako prywatne pole InitialNumberOfTries równy trzy, i zrobiła je const. A już lokalnie dla metody zrobiłabym zmienną "currentNumberOfTries". Dzięki temu klasa nadal będzie stateless (bo stałe pole to nie jest stan klasy). Ogólnie bardzo nie lubię klas stateful, w których wartości pól się zmieniają. No chyba że klasa z definicji jest strukturą danych, ale to co innego trochę.

Patrzę zresztą na mój kod i widzę że dokładnie tak to zrobiłam :D.
![image](https://user-images.githubusercontent.com/6991707/204283864-9ae626e2-0b0c-477d-8650-2aacdcc76853.png)

(ciekawe czy się pokaże obrazek czy dziwny link, bo ja na razie widzę dziwny link). Jak widać u mnie 6, czyli liczba boków kostki, jest w osobnej klasie, co chyba jest lepszym designem jednak.

Reply:
No tak u mnie było w zalozeniu, że liczba prób to cecha gry, a próby gracza liczyłam counterem, ktory teraz przechrzcilam na triesUsed żeby był mniej generyczny niż "counter".
(tak te obrazki sie wklejaja, to jest jakis link do githubowej bazki obrazków chyba)